src/B4580.c

```diff
@@ -610,30 +727,212 @@ void render_animated_model_with_vertices(s32 animatorID, Mtx* rootTransform, s32
 
         animatorID &= ~0x800;
         animator = (*gCurrentAnimMeshListPtr)[animatorID];
-        if (animator != NULL && animator->flags != 0 && !(animator->flags & MODEL_ANIMATOR_FLAGS_40) &&
-            animator->flags & (1 << gCurrentCamID) && !(animator->flags & MODEL_ANIMATOR_FLAGS_80))
+        if (animator != NULL && animator->flags != 0 && !(animator->flags & MODEL_ANIMATOR_FLAGS_UPDATE_PENDING) &&
+            animator->flags & (1 << gCurrentCamID) && !(animator->flags & MODEL_ANIMATOR_FLAGS_HIDDEN))
         {
             animator->mtx = *rootTransform;
-            D_80153A60 = vtxSegment;
+            gAnimVtxSegment = vtxSegment;
             animator->vertexArray = vertexArray;
             rtPtr->appendGfxArg = animator;
-            rtPtr->appendGfx = &appendGfx_animator;
+            rtPtr->appendGfx = (void (*)(void*))appendGfx_animator;
             rtPtr->distance = 0;
             rtPtr->renderMode = animator->renderMode;
             queue_render_task(rtPtr);
         }
     }
 }
 
-INCLUDE_ASM(s32, "B4580", appendGfx_animator);
+void appendGfx_animator(ModelAnimator* animator) {
+    Matrix4f sp10;
+
+    if (animator->vertexArray != NULL) {
+        gSPSegment(gMasterGfxPos++, gAnimVtxSegment, VIRTUAL_TO_PHYSICAL(animator->vertexArray));
+    }
+
+    gDisplayContext->matrixStack[gMatrixListPos] = animator->mtx;
+    gSPMatrix(gMasterGfxPos++, &gDisplayContext->matrixStack[gMatrixListPos++], G_MTX_PUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    switch (gAnimModelFogEnabled != 0) {
```

Comment:
Can this be an if/else? A switch is a bit awkward here imo

Reply:
couldn't match it with if/else, so i left TODO comment